internal/service/amplify/app_test.go

```diff
@@ -324,6 +325,51 @@ func testAccApp_CacheConfig(t *testing.T) {
 	})
 }
 
+func testAccApp_ComputeRole(t *testing.T) {
+	ctx := acctest.Context(t)
+	var app1, app2, app3 types.App
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	resourceName := "aws_amplify_app.test"
+	computeRole1ResourceName := "aws_iam_role.compute_test1"
+	computeRole2ResourceName := "aws_iam_role.compute_test2"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(ctx, t); testAccPreCheck(t) },
+		ErrorCheck:               acctest.ErrorCheck(t, names.AmplifyServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckAppDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAppConfig_computeRoleARN(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAppExists(ctx, resourceName, &app1),
+					resource.TestCheckResourceAttrPair(resourceName, "compute_role_arn", computeRole1ResourceName, names.AttrARN)),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAppConfig_computeRoleARNUpdated(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAppExists(ctx, resourceName, &app2),
+					testAccCheckAppNotRecreated(&app1, &app2),
```

Comment:
Can this be changed to an [`ExpectResourceAction` plancheck](https://developer.hashicorp.com/terraform/plugin/testing/acceptance-tests/plan-checks/resource#expectresourceaction-plan-check)?

Reply:
Yes, done.